awscrt/http.py

```diff
@@ -127,14 +150,18 @@ def new(cls,
             if not bootstrap:
                 bootstrap = ClientBootstrap.get_or_create_static_default()
 
-            connection = cls()
+            connection = expected_cls()
             connection._host_name = host_name
             connection._port = port
 
             def on_connection_setup(binding, error_code, http_version):
                 if error_code == 0:
                     connection._binding = binding
                     connection._version = HttpVersion(http_version)
+                    if expected_version and expected_version != connection._version:
+                        # unexpected protocol version
```

Comment:
Currently, `expected_version` is always None

Reply:
If Http2ClientConnection.new() is always supposed to create an HTTP/2 connection, shouldn't it do things to try and guarantee that's the kind of connection which gets made? Like set the ALPN string, or prior_knowledge?